### PR TITLE
[MRG] FIX Topographic plotting for KIT-UMD data

### PR DESCRIPTION
**Pull request by christianbrodbeck**
Related to https://github.com/mne-tools/mne-python/issues/3311. 

The builtin KIT-157 layout is based on the KIT-NY system and does not apply for KIT-UMD data. This PR distinguishes the two systems based on sensor positions. If in the future the KIT system-identifier is stored in the measurement info this distinction will be more efficient, but the sensor-based distinction will always be useful for legacy FIFF files.

### CODE REVIEW

**Comment by dengemann** [comment]
Give a sign once it's ready.

**Comment by christianbrodbeck** [comment]
This is ready as a temporary fix that makes plotting work for UMD data.

**Comment by christianbrodbeck** [comment]
The travis problem is just the `FAIL: Doctest: mne.filter.construct_iir_filter`

**Comment by agramfort** [comment]
doesn't it break/change the code of people's code using KIT at NYU?

File: mne/channels/layout.py [hunk 1]
```diff
@@ -446,7 +446,9 @@ def find_layout(info, ch_type=None, exclude='bads'):
     elif has_CTF_grad:
         layout_name = 'CTF-275'
     elif n_kit_grads <= 157:
-        layout_name = 'KIT-157'
+        # This applies to the KIT systems at NYU and UMD which have different
+        # layouts. The 'KIT-157' layout applies to the NYU system.
+        return None
```
Comment (jaeilepp) [inline mne/channels/layout.py]: This doesn't look right. Like this it would never find 'KIT-157'. It's still in use somewhere, right?

Comment (christianbrodbeck) [inline mne/channels/layout.py]: You can still manually specify it.

**Comment by christianbrodbeck** [comment]
@agramfort it will change the looks of topographic plots slightly because the layout will be slightly different. To avoid that (and still make it work for UMD data) I would have to add the proposed guessing of system by sensor positions.

**Comment by christianbrodbeck** [comment]
Actually I noticed not all topographic plotting functions currently resort to the automatic layout, so we'll need a more involved fix for this...

**Comment by agramfort** [comment]
that's what I feared

**Comment by christianbrodbeck** [comment]
This new solution should not break any old code, and can be updated to be more efficient once we have KIT System-IDs in the measurement info.

File: mne/channels/layout.py [hunk 2]
```diff
@@ -445,10 +445,10 @@ def find_layout(info, ch_type=None, exclude='bads'):
...
     elif has_CTF_grad:
         layout_name = 'CTF-275'
-    elif n_kit_grads <= 157:
-        layout_name = 'KIT-157'
-    elif n_kit_grads > 157:
-        layout_name = 'KIT-AD'
+    elif n_kit_grads > 0:
+        layout_name = _find_kit_system(info, n_kit_grads)
+        if layout_name == 'KIT-UMD':
+            return _auto_layout(info)
```
Comment (agramfort) [inline mne/channels/layout.py]: why not adding a layout file for KIT-UMD? by doing this we could share the file between software.

Comment (christianbrodbeck) [inline mne/channels/layout.py]: Ho would I create a layout file, using the `generate_2d_layout` function? And what are the criteria for a good layout? That `plot_topo` looks good?

Comment (agramfort) [inline mne/channels/layout.py]: if generate_2d_layout output looks good with plot_topo yes.

Comment (christianbrodbeck) [inline mne/channels/layout.py]: Ok, I generated an automatic layout and then I nudged the rectangles around manually to minimize overlap, which seemed preferable for plot_topo. I noticed though that topomap also uses the same layout where a deterministic arrangement seems preferable?

kit1
kit4

Comment (christianbrodbeck) [inline mne/channels/layout.py]: There is some, but limited symmetry... here is a `raw.plot_sensors()` plot and a layout generated with MNE `_auto_topomap_coords()`:

sensors

auto

Comment (agramfort) [inline mne/channels/layout.py]: ok indeed. Then I think you're manually edited version is good enough.

**Comment by christianbrodbeck** [comment]
@agramfort I added a UMD layout. It will be easier to add a test via the KIT_SYSTEM_ID PR because it adds a small UMD raw file.

**Comment by agramfort** [comment]
thx @christianbrodbeck

**Pull request merged**